ksql-parser/src/main/java/io/confluent/ksql/parser/tree/TableElements.java

```diff
@@ -114,17 +113,6 @@ private static TableElements build(final Stream<TableElement> elements) {
     throwOnDuplicateNames(keyColumns, "KEY");
     throwOnDuplicateNames(valueColumns, "non-KEY");
 
-    final long numKeyColumns = keyColumns.size();
```

Comment:
These checks here are not needed?

Reply:
That's why I deleted them ;) :p

So the PR moved these checks, and the ones in `TableElement` out of the parser and into the command factory.   When you combine the checks it's actually the case that these checks are superfluous - the checks in `TableElement` mean its not possible for the data to be in the state these checks detect.